BusinessLogicLayer/Initializer/Implementation/Initialize.cs

```diff
@@ -0,0 +1,123 @@
+using DataAccessLayer.DataBaseModels;
+using Microsoft.EntityFrameworkCore.Metadata;
+using OfficeOpenXml;
+
+namespace BusinessLogicLayer.Initializer.Implementation
+{
+    public class Initialize : IIInitialize
+    {
+        private UnitOfWork _unitOfWork = new UnitOfWork();
+
+        public string CreatePath(string directoryName = "", string fileName = "Languages")
+        {
+            string rootPath = Directory.GetCurrentDirectory() + @"\wwwroot\Dictionary\" + directoryName + fileName + ".xlsx";
+            Console.WriteLine(rootPath);
+            return rootPath;
+        }
+
+        public ExcelWorksheet GetDataFromFile(string path)
+        {
+            FileInfo fileInfo = new FileInfo(path);
+            ExcelPackage.LicenseContext = LicenseContext.NonCommercial;
+            ExcelPackage package = new ExcelPackage(fileInfo);
+            ExcelWorksheet worksheet = package.Workbook.Worksheets.FirstOrDefault();
+            return worksheet;
+        }
+
+        public List<ParsedData> ParseData(ExcelWorksheet worksheet)
+        {
+            var parsedData = new List<ParsedData>();
+
+            if (worksheet != null)
+            {
+                // get number of rows and columns in the sheet
+                int rows = worksheet.Dimension.Rows;
+                int columns = worksheet.Dimension.Columns;
+
+                // loop through the worksheet rows and columns
+                for (int i = 2; i <= rows; i++)
+                {
+                    for (int j = 2; j <= columns; j++)
+                    {
+                        if (worksheet.Cells[i, j].Value == null) 
+                            break;
+
+                        var data = new ParsedData
+                        {
+                            Word = worksheet.Cells[i, 1].Value.ToString(),
+                            Language = worksheet.Cells[1, j].Value.ToString(),
+                            Translation = worksheet.Cells[i, j].Value.ToString()
+                        };
+                        parsedData.Add(data);
+                    }
+                }
+            }
+
+            return parsedData;
+        }
+
+        public void WriteLanguageDataToDataBase()
+        {
+            //Write data to languages table
+            List<ParsedData> languageData = ParseData(GetDataFromFile(CreatePath()));
+            List<Language> allLanguages = new List<Language>();
+            for (int i = 0, j = 0; i < languageData.Count; i++)
+            {
+                if (allLanguages.FirstOrDefault(l => l.FullName == languageData[i].Word) == null)
+                {
+                    allLanguages.Add(new Language()
+                    {
+                        ShortName = languageData[i + j].Language,
+                        FullName = languageData[i + j++].Word
+                    });
+                }
+            }
+
+            //Save data to dataBase
+            foreach (var item in allLanguages)
```

Comment:
Not the best approach for several reasons:
1) you have DAL logic in BLL
2) you make several DB requests in cycle - and this is very bad from DB performance POV, because you have to open and close a new connection each time. Instead of this you can write an additional DAL method, that will save many entries in one time. But here you meet another limitation - the IRepository pattern is broken. My recommendation here is to remove this pattern, because it limits you a lot with the list of actions you can do with DB entry.